### PR TITLE
change keycode structure in get_key()

### PR DESCRIPTION
**Pull request by borstel**
- make function get_key() work on big endian machines

### CODE REVIEW

**Comment by mptre** [comment]
Looks good, but I would not remove the `length` field and the inclusion
of `sys/ttydefaults.h` can now be removed.

**Comment by calleerlandsson** [comment]
Hey @borstel, thanks for contributing!

This looks good to me. Would you mind rebasing your branch on the current master?

File: src/pick.c [hunk 1]
```diff
@@ -822,3 +817,4 @@ isu8start(unsigned char c)
 {
 	return (c & (0x80 | 0x40)) == (0x80 | 0x40);
 }
+
```
Comment (mptre) [inline src/pick.c]: Remove blank line.

**Comment by mptre** [comment]
Besides one minor comment on the latest commit this looks good. How
about squashing the commits after rebasing?

**Comment by mptre** [comment]
Sorry for the bike shedding but I think we should remove the key variable
completely instead:

```diff
diff --git a/src/pick.c b/src/pick.c
index bc43fbe..7efc5f8 100644
--- a/src/pick.c
+++ b/src/pick.c
@@ -746,15 +746,14 @@ get_key(char *buf, size_t size, size_t *nread)
 		{ "\033O3~",	4,	DEL },
 		{ NULL,		0,	0 },
 	};
-	const char	*key;
 	int		 i;
 
 	*nread = 0;
 getc:
 	buf[(*nread)++] = tty_getc();
 	size--;
-	for (i = 0; (key = keys[i].s); i++) {
-		if (*nread > keys[i].length || strncmp(buf, key, *nread))
+	for (i = 0; keys[i].s != NULL; i++) {
+		if (*nread > keys[i].length || strncmp(buf, keys[i].s, *nread))
 			continue;
 
 		if (*nread == keys[i].length)
```

**Comment by mptre** [comment]
Looks great by now!

**Comment by borstel** [comment]
Thank you for suggesting improvements.

**Comment by mptre** [comment]
Would you mind rebasing your changes once again?

**Comment by calleerlandsson** [comment]
Thanks @borstel! I'll merge this.

**Pull request merged**